guides/source/association_basics.md

```diff
@@ -761,15 +761,53 @@ class Book < ApplicationRecord
 end
 ```
 
-Active Record will attempt to automatically identify that these two models share a bi-directional association based on the association name. In this way, Active Record will only load one copy of the `Author` object, making your application more efficient and preventing inconsistent data:
+Active Record will attempt to automatically identify that these two models share
+a bi-directional association based on the association name. This information
+allows Active Record to:
+
+* Prevent needless queries for already-loaded data
+
+```irb
+irb> author = Author.first
+irb> author.books.all? do |book|
+irb>   book.author.equal?(author) # No additional queries executed here
+irb> end
+=> true
+```
+
+* Prevent inconsistent data (since there is only one copy of the `Author` object
+  loaded)
 
 ```irb
-irb> a = Author.first
-irb> b = a.books.first
-irb> a.first_name == b.author.first_name
+irb> author = Author.first
+irb> book = author.books.first
+irb> author.name == book.author.name
 => true
-irb> a.first_name = 'David'
-irb> a.first_name == b.author.first_name
+irb> author.name = 'Changed Name'
+irb> author.name == book.author.name
+=> true
+```
+
+* Autosave associations in more cases
+
+```irb
+irb> author = Author.new
+irb> book = author.books.new
+irb> book.save!
+irb> book.persisted?
+=> true
+irb> author.persisted?
+=> true
+```
+
+* Validate the [presence](active_record_validations.html#presence) and [absence](active_record_validations.html#absence) of associations in more cases
+
+```irb
+irb> author = Author.new
+irb> book = author.books.new
+irb> book.author.present?
+=> true
+irb> book.valid?
 => true
```

Comment:
Perhaps a counter example here, to further illustrate the point?

```suggestion
irb> book = Book.new
irb> book.author.present?
=> false
irb> book.valid?
=> false
irb> author = Author.new
irb> book = author.books.new
irb> book.author.present?
=> true
irb> book.valid?
=> true
```

Or maybe with validation error messages instead?

```suggestion
irb> book = Book.new
irb> book.valid?
=> false
irb> book.errors.full_messages
=> ["Author must exist"]
irb> author = Author.new
irb> book = author.books.new
irb> book.valid?
=> true
```